[Solomon0519] iP

Solomon0519 opened the pull request:
#  **Luffy Bot**

## Luffy Bot is a Chat Bot made by me, built on ***many tears wept through many nights***.
> Luffy Bot is always improving!
> Luffy Bot updates based on its [users' needs!](https://nus-cs2103-ay2425s1.github.io/website/index.html)

Luffy Bot can do:

- Saving your tasks!
- Giving your tasks a deadline!
- Marking your tasks! 

Luffy Bot is:

1.  Suitable for all ages
2.  Easy to use
3.  Trying to become like ChatGPT :100:

~~~ 
Luffy Bot is completely free for all!
Its source code is free and available on Github!
~~~

**Features**:

- [x] Saves and loads tasks.
- [x] Add and delete tasks.
- [x] Mark and unmark tasks.
- [ ] Better parsing and formatting of task list (Work in Progress)

Luffy Bot has many lines of `code` and I hope nothing crashes. :smile_cat:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Solomon0519 committed:
Testing

Solomon0519 committed:
Changes

Solomon0519 committed:
Testing

Solomon0519 committed:
Merge remote-tracking branch 'origin/master'

Solomon0519 committed:
Level-1 Changes

Solomon0519 committed:
Level-1 Changes

Solomon0519 committed:
Level-1 Changes

Solomon0519 committed:
Level-4 Changes

Solomon0519 committed:
Level-5 Changes

Solomon0519 committed:
Level-6 Changes

Solomon0519 committed:
Test Changes for Week 3

Solomon0519 committed:
Test Changes for Week 3

Solomon0519 committed:
Test Changes for Week 3

Solomon0519 committed:
Merge branch 'branch-Level-7'

Solomon0519 committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

Solomon0519 committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

Solomon0519 committed:
Commit Level-7 Branch

Solomon0519 committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

Solomon0519 committed:
Merge remote-tracking branch 'origin/branch-Level-7' into branch-Level-7

Solomon0519 committed:
Merge branch 'branch-Level-7'

Solomon0519 committed:
Revert changes

Solomon0519 committed:
Revert changes

Solomon0519 committed:
Help me please

Solomon0519 committed:
Merge branch 'master' of https://github.com/Solomon0519/ip

Solomon0519 committed:
Merge pull request #1 from Solomon0519/branch-Level-8

Branch level 8

Solomon0519 committed:
Merge 8 to Master

Solomon0519 committed:
Merge branch 'master' into branch-Level-8

Solomon0519 committed:
Revert changes

Solomon0519 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Luffy.java

Solomon0519 committed:
A-Packages

Solomon0519 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Solomon0519 committed:
WHY IS NOTHING WORKING

Solomon0519 committed:
Revert "WHY IS NOTHING WORKING"

This reverts commit e030dfc77a9a6f2df2d7843c087c5e5676be10be.

Solomon0519 committed:
OK ITS BACK NOW

Solomon0519 committed:
Loaded Gradle

Solomon0519 committed:
Amended all merges

Solomon0519 committed:
Merge remote-tracking branch 'origin/master'

Solomon0519 committed:
A-JavaDoc

Solomon0519 committed:
Fix file writing issue

Solomon0519 committed:
Level-9

Solomon0519 committed:
Checkstyle

Solomon0519 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/luffy/FindCommand.java
#	src/main/java/luffy/LuffyParser.java

Solomon0519 committed:
Edit build.gradle

Solomon0519 committed:
Add A-JUnit Tests

Solomon0519 committed:
A-Jar

RezwanAhmed123 commented on src/main/java/luffy/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Override
+    public String dataToSave() {
```
I did something similar in my code as you did here! Maybe the method could be named a little more intuitively so that it may make the code easier to understand?
```suggestion
    public String formatToStorageData() {
```

Solomon0519 replied:
Yep, this will make the methods more understandable, thank you!

RezwanAhmed123 commented on src/main/java/luffy/DeleteCommand.java:
```diff
@@ -0,0 +1,35 @@
+     * @param taskList array list of existing tasks
+     */
+    @Override
+    public void executeCmd(LuffyUI ui, Storage taskStorage, TaskList taskList) {
```
I like this method name! Very succinct and understandable!

Solomon0519 replied:
Thank you!

RezwanAhmed123 commented on src/main/java/luffy/Luffy.java:
```diff
@@ -0,0 +1,45 @@
+ * Represents a chatbot that has various commands
+ * for users to note down tasks and check them off
+ */
+public class Luffy {
```
Peak character, therefore a good choice of bot name! :)

Solomon0519 replied:
Hahaha, peak anime!

RezwanAhmed123 commented on src/main/java/luffy/LuffyParser.java:
```diff
@@ -0,0 +1,113 @@
+     * @throws LuffyException If the expectedLength does not equal to the length of the textList
+     */
+    public static void checkValidArguments(String[] textList, int expectedLength) throws LuffyException {
+        if (textList.length != expectedLength && expectedLength == 2) {
+            throw new LuffyException("Your deadline task requires a deadline!");
+        } else if (textList.length != expectedLength && expectedLength == 3) {
+            throw new LuffyException("Your event task requires a start time and end time!");
+        }
```
Good catch! Its always good to ensure there is exactly 2 or 3 commands because more or less than required commands could be given for either command.

RezwanAhmed123 commented on src/main/java/luffy/LuffyUI.java:
```diff
@@ -0,0 +1,128 @@
+     */
+    public void showWelcomeMessage() {
+        showLine();
+        System.out.println(" Hello! I'm luffy.Luffy");
```
Perhaps a possible typo?

```suggestion
        System.out.println(" Hello! I'm Luffy");
```

yyihaoc replied:
^ I had the same problem when refactoring it into packages and classes as IDEs might automatically search for name occurence and modify then during the process.

RezwanAhmed123 reviewed (commented):
LGTM, some nits. Clarity of code is great!

yyihaoc commented on src/main/java/luffy/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+import java.io.IOException;
+
+/**
+ * Represents a command that adds a task into the list of tasks
```
Should this end with a period?

yyihaoc commented on src/main/java/luffy/Todo.java:
```diff
@@ -0,0 +1,29 @@
+    @Override
+    public String dataToSave() {
+        if (checkIsDone()) {
+            return String.format("TO-DO         | 1 | %s", super.getTaskInfo());
```
I like how you made the string format easy to see in the data file by making the divider aligned for various tasks.

yyihaoc commented on src/main/java/luffy/Task.java:
```diff
@@ -0,0 +1,79 @@
+     */
+    public Task(String taskInfo, boolean isDone) {
+        this.taskInfo = taskInfo;
+        this.isDone = isDone;
```
Perhaps you don't need `isDone` since all tasks recorded are not done at first.

yyihaoc commented on src/main/java/luffy/UnmarkCommand.java:
```diff
@@ -0,0 +1,40 @@
+        this.markIndex = index;
+    }
+
+    /**
+     * This method adds an executable command to unmark a
+     * task in the task list at a specific index by the UI
+     *
+     * @param ui user interface for Chat Bot
+     * @param taskStorage storage location for file
+     * @param taskList array list of existing tasks
+     */
```
I like how comprehensive the descriptions are! Same with the others!

yyihaoc commented on src/main/java/luffy/scratchfile.txt:
```diff
@@ -0,0 +1,139 @@
+//            }
+//
+//            if (echoPhrase.equals("bye")) {
+//
```
Line breaks between code blocks made it easier to see. Perhaps within code blocks it's better to keep them together for easier review?

yyihaoc reviewed (commented):
LGTM! Just a few nits to fix!

Solomon0519 committed:
Level-10 branch

Solomon0519 committed:
Fix GUI by factoring functions

Solomon0519 committed:
Refactor files to make searching easier

Solomon0519 committed:
Insert assertions for important checks

User input may be empty sometimes, need to check for marking, unmarking and deleting.

Solomon0519 committed:
Merge pull request #2 from Solomon0519/branch-A-Assertions

Insert assertions for important checks

Solomon0519 committed:
Add documentation to more methods

Make clearer the use of functions and increase readability of code.

Solomon0519 committed:
Merge pull request #3 from Solomon0519/branch-A-CodeQuality

Add documentation to more methods

Solomon0519 committed:
Add function for mass marking

Users may want to mark more than one task in each command.

Solomon0519 committed:
Merge pull request #4 from Solomon0519/C-MassOps

Add function for mass marking

Solomon0519 committed:
Change file system so that it tackles file not file

Previous iterations forget to take into account that other computers may not have the file.

Rewrite code to create a directory and new file in their local computer.

Solomon0519 committed:
Merge pull request #6 from Solomon0519/C-MassOps

Change file system so that it tackles file not file

Solomon0519 committed:
Add some code to make logic better

Solomon0519 committed:
Merge pull request #7 from Solomon0519/C-MassOps

Add some code to make logic better

Solomon0519 committed:
Edit README.md for user guide

Solomon0519 committed:
Finalise the bot

Solomon0519 committed:
Finalise the minor display issues
